src/rospkg/environment.py

```diff
@@ -109,7 +109,7 @@ def get_ros_package_path(env=None):
     """
     if env is None:
         env = os.environ
-    return env.get(ROS_PACKAGE_PATH, None)
+    return env.get(ROS_PACKAGE_PATH, '') + ':/usr/share'
```

Comment:
> Can you give an example where this could break so we can find a fix for that?

Sure, I think I can demonstrate how it can be logically inconsistent, and a come up with a hypothetical scenario where that inconsistency could actually actually break something.

If you have packages A, B, and C installed from Ubuntu (into `/usr`) and you have A, B, and D installed in `/opt/ros/kinetic`, then you do `rospack list` you'll get A, B, and C. However, if you source the `/opt/ros/kinetic/setup.bash` and do `rospack list` you'll get A, B, C, and D. Where C comes from `/usr` and the others point to `/opt/ros/kinetic`. Let's say that both C and D build against (use headers from and link against) B.

This can be a problem if, for example, you build a package locally called E which uses both C and D.

First consider you do this in a clean shell. The build should fail because D is not found.

Then consider you try to build it after sourcing the `/opt/ros/kinetic/setup.bash` file. Then the build does not fail because it finds C in `/usr` and D in `/opt/ros/kinetic`. The "expected" behavior would be that the build fails because C is not found, but that's not what happens. However, even if the build works (might not due to differences in headers between versions of B), it very likely will have problems when running since C is built against B in `/usr` and D is built against B in `/opt/ros/kinetic` (let's call that B'). If B and B' are not ABI compatible, then you've got a problem.

This same issue can be extended to Python code, cmake modules, locating resources from other packages, etc.

---

There is also another issue that comes up that's sort of the reverse of the above, where you get the `rospkg` from OSRF rather than Ubuntu because it's newer (for example). That leads to failures to run things like `roslaunch` when you have ROS installed from both places. I think I sent you emails on this issue, but running `roslaunch` I got:

```
william@ubuntu:~$ roscore
WARNING: unable to configure logging. No log files will be generated
Checking log directory for disk usage. This may take awhile.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/roslaunch/__init__.py", line 307, in main
    p.start()
  File "/usr/lib/python2.7/dist-packages/roslaunch/parent.py", line 268, in start
    self._start_infrastructure()
  File "/usr/lib/python2.7/dist-packages/roslaunch/parent.py", line 217, in _start_infrastructure
    self._load_config()
  File "/usr/lib/python2.7/dist-packages/roslaunch/parent.py", line 132, in _load_config
    roslaunch_strs=self.roslaunch_strs, verbose=self.verbose)
  File "/usr/lib/python2.7/dist-packages/roslaunch/config.py", line 445, in load_config_default
    load_roscore(loader, config, verbose=verbose)
  File "/usr/lib/python2.7/dist-packages/roslaunch/config.py", line 92, in load_roscore
    f_roscore = get_roscore_filename()
  File "/usr/lib/python2.7/dist-packages/roslaunch/config.py", line 84, in get_roscore_filename
    return os.path.join(r.get_path('roslaunch'), 'resources', 'roscore.xml')
  File "/usr/lib/python2.7/dist-packages/rospkg/rospack.py", line 200, in get_path
    raise ResourceNotFound(name, ros_paths=self._ros_paths)
ResourceNotFound: roslaunch
```

This happened because the version of `rospkg` that got installed didn't have this patch and so it couldn't find `roslaunch` in `/usr`. That's why I was so interested in finding a way to satisfy the Debian use case with a patch that we could merge upstream -- to avoid this issue.

---

For both of these reasons we've resorted to recommending people not install both at the same time. In practice this is probably fine, but not an ideal outcome.